Git-commit: a2bf1d1a4bd8a22d8a71e760e84926bc47d6c2a2
From: Chiqijun <chiqijun@huawei.com>
Date: Tue, 2 Mar 2021 20:07:44 +0800
Subject: [PATCH] net/hinic: Remove the initialization of the global variable
 g_uld_info
Patch-mainline: Queued in openEuler repo, version 5.10.0-4.5.0
References: openEuler-21.03
Git-repo: https://gitee.com/openeuler/kernel.git

driver inclusion
category: bugfix
bugzilla: 47993

-----------------------------------------------------------------------

In the hinic driver, the g_uld_info global variable is assigned to all
0s, and the'designated init' alarm is reported at compile time. The
global variable defaults to all 0s, and the assignment of 0s is not
necessary.

Signed-off-by: Chiqijun <chiqijun@huawei.com>
Reviewed-by: Wangxiaoyun <cloud.wangxiaoyun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 drivers/net/ethernet/huawei/hinic/hinic_lld.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/huawei/hinic/hinic_lld.c b/drivers/net/ethernet/huawei/hinic/hinic_lld.c
index fddb949ff9aa..6c960cecf101 100644
--- a/drivers/net/ethernet/huawei/hinic/hinic_lld.c
+++ b/drivers/net/ethernet/huawei/hinic/hinic_lld.c
@@ -130,7 +130,7 @@ struct hinic_pcidev {
 #define MAX_CARD_ID 64
 static u64 card_bit_map;
 LIST_HEAD(g_hinic_chip_list);
-struct hinic_uld_info g_uld_info[SERVICE_T_MAX] = { {0} };
+struct hinic_uld_info g_uld_info[SERVICE_T_MAX];
 static const char *s_uld_name[SERVICE_T_MAX] = {
 	"nic", "ovs", "roce", "toe", "iwarp", "fc", "fcoe", "migrate"};
 
-- 
2.31.1

